functions/src/resolve-market.ts

```diff
@@ -15,156 +15,146 @@ import {
-        return { status: 'error', message: 'Invalid value' }
-
-      if (
-        outcomeType === 'BINARY' &&
-        probabilityInt !== undefined &&
-        (probabilityInt < 0 ||
-          probabilityInt > 100 ||
-          !isFinite(probabilityInt))
-      )
-        return { status: 'error', message: 'Invalid probability' }
-
-      if (creatorId !== userId)
-        return { status: 'error', message: 'User not creator of contract' }
+  if (creatorId !== userId)
+    throw new APIError(403, 'User is not creator of contract')
 
-      if (contract.resolution)
-        return { status: 'error', message: 'Contract already resolved' }
+  if (contract.resolution) throw new APIError(400, 'Contract already resolved')
 
-      const creator = await getUser(creatorId)
-      if (!creator) return { status: 'error', message: 'Creator not found' }
+  const creator = await getUser(creatorId)
+  if (!creator) throw new APIError(400, 'Creator not found')
 
-      const resolutionProbability =
-        probabilityInt !== undefined ? probabilityInt / 100 : undefined
+  const resolutionProbability =
+    probabilityInt !== undefined ? probabilityInt / 100 : undefined
 
-      const resolutionTime = Date.now()
-      const newCloseTime = closeTime
-        ? Math.min(closeTime, resolutionTime)
-        : closeTime
+  const resolutionTime = Date.now()
+  const newCloseTime = closeTime
+    ? Math.min(closeTime, resolutionTime)
+    : closeTime
 
-      const betsSnap = await firestore
-        .collection(`contracts/${contractId}/bets`)
-        .get()
+  const betsSnap = await firestore
+    .collection(`contracts/${contractId}/bets`)
+    .get()
 
-      const bets = betsSnap.docs.map((doc) => doc.data() as Bet)
+  const bets = betsSnap.docs.map((doc) => doc.data() as Bet)
 
-      const liquiditiesSnap = await firestore
-        .collection(`contracts/${contractId}/liquidity`)
-        .get()
+  const liquiditiesSnap = await firestore
+    .collection(`contracts/${contractId}/liquidity`)
+    .get()
 
-      const liquidities = liquiditiesSnap.docs.map(
-        (doc) => doc.data() as LiquidityProvision
-      )
+  const liquidities = liquiditiesSnap.docs.map(
+    (doc) => doc.data() as LiquidityProvision
+  )
 
-      const { payouts, creatorPayout, liquidityPayouts, collectedFees } =
-        getPayouts(
-          outcome,
-          resolutions ?? {},
-          contract,
-          bets,
-          liquidities,
-          resolutionProbability
-        )
-
-      await contractDoc.update(
-        removeUndefinedProps({
-          isResolved: true,
-          resolution: outcome,
-          resolutionValue: value,
-          resolutionTime,
-          closeTime: newCloseTime,
-          resolutionProbability,
-          resolutions,
-          collectedFees,
-        })
-      )
+  const { payouts, creatorPayout, liquidityPayouts, collectedFees } =
+    getPayouts(
+      outcome,
+      Object.fromEntries(resolutions || []),
```

Comment:
Kind of awkward to parse into a `Map` and turn it into an object here. Why not have the schema parse it into an object?